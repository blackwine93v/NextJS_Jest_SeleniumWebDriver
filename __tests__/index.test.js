
import { shallow } from 'enzyme'
import React from 'react'
import renderer from 'react-test-renderer'

import App from '../pages/index.js'

import webdriver from 'selenium-webdriver'
import homeObject from '../e2e/tests/pages/home'

const By = webdriver.By
const driver = new webdriver.Builder()
  .forBrowser('firefox')
  .build();
const Home = homeObject(driver)

describe('Input name with e2e test', () => {
  beforeAll((done) => {
    return Home.navigate()
      .then(() => { done() })
  });

  afterAll(() => {
    driver.quit()
      .then(() => done())
  })

  it('app shows "Hello, VuongTEST"', () => {
    Home.enterName('TEST')
    return Home.getName()
      .then(inputValue => expect(inputValue).toBe('Hello, VuongTEST'));
  })

  it('name should be in blue', () => {
    return Home.getNameColor()
      .then(color => expect(color).toBe('rgb(0, 0, 255)'));
  })
})

describe('With Enzyme', () => {
  it('App shows "Hello, Vuong"', () => {
    const app = shallow(<App />)
    expect(app.find('p').text()).toEqual('Hello, Vuong')
  })
})

describe('With Snapshot Testing', () => {
  it('App shows "Hello, Vuong"', () => {
    const component = renderer.create(<App />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})