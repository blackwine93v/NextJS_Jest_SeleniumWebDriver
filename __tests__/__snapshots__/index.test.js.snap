// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing App shows "Hello, Vuong" 1`] = `
<div>
  <header>
    This is header
  </header>
  <h3
    className="jsx-1405864837"
  >
    Home
  </h3>
  <p
    className="jsx-1405864837 name"
  >
    Hello, 
    Vuong
  </p>
  <br
    className="jsx-1405864837"
  />
  <input
    className="jsx-1405864837 nameInput"
    onChange={[Function]}
    type="input"
    value="Vuong"
  />
  <footer>
    This is footer
  </footer>
</div>
`;
